src/symbols.c

```diff
@@ -1914,27 +1867,276 @@ static void load_user_tags(filetype_id ft_id)
 }
 
 
+static void on_focus_out(GtkWidget *list, GdkEventFocus *unused, gpointer *user_data)
+{
+	gtk_widget_hide(tag_goto_popup);
+}
+
+
+static void on_row_activated(GtkTreeView *tree_view, GtkTreePath *path,
+	GtkTreeViewColumn *column, gpointer user_data)
+{
+	GtkTreeModel *model = gtk_tree_view_get_model(tree_view);
+	GeanyDocument *new_doc;
+	GtkTreeIter iter;
+	TMTag *tag;
+
+	gtk_tree_model_get_iter(model, &iter, path);
+	gtk_tree_model_get(model, &iter, TAG_COLUMN, &tag, -1);
+	g_return_if_fail(tag);
+
+	new_doc = document_find_by_real_path(tag->file->file_name);
+	if (!new_doc)
+		new_doc = document_open_file(tag->file->file_name, FALSE, NULL, NULL);
+
+	if (new_doc)
+	{
+		GeanyDocument *old_doc = document_get_current();
+
+		navqueue_goto_line(old_doc, new_doc, tag->line);
+	}
+
+	gtk_widget_hide(tag_goto_popup);
+	tm_tag_unref(tag);
+}
+
+
+static gboolean on_key_pressed(GtkWidget *widget, GdkEventKey *event, gpointer user_data)
+{
+	guint state = event->state & gtk_accelerator_get_default_mod_mask();
+
+	if (event->keyval == GDK_Escape && state == 0)
+	{
+		gtk_widget_hide(tag_goto_popup);
+		keybindings_send_command(GEANY_KEY_GROUP_FOCUS, GEANY_KEYS_FOCUS_EDITOR);
+		return TRUE;
+	}
+	return FALSE;
+}
+
+
+static const gchar *get_tag_class(const TMTag *tag)
+{
+	switch (tag->type)
+	{
+		case tm_tag_prototype_t:
+		case tm_tag_method_t:
+		case tm_tag_function_t:
+			return "classviewer-method";
+		case tm_tag_variable_t:
+		case tm_tag_externvar_t:
+			return "classviewer-var";
+		case tm_tag_macro_t:
+		case tm_tag_macro_with_arg_t:
+			return "classviewer-macro";
+		case tm_tag_class_t:
+			return "classviewer-class";
+		case tm_tag_member_t:
+		case tm_tag_field_t:
+			return "classviewer-member";
+		case tm_tag_typedef_t:
+		case tm_tag_enum_t:
+		case tm_tag_union_t:
+		case tm_tag_struct_t:
+			return "classviewer-struct";
+		case tm_tag_namespace_t:
+		case tm_tag_package_t:
+			return "classviewer-namespace";
+		default:
+			break;
+	}
+	return "classviewer-struct";
+}
+
+
+static void create_goto_popup(void)
+{
+	GtkWidget *frame, *scroller;
+	GtkListStore *store;
+	GtkTreeSelection *selection;
+	GtkTreeViewColumn *column;
+	GtkCellRenderer *renderer;
+
+	tag_goto_popup = g_object_new(GTK_TYPE_WINDOW, "type", GTK_WINDOW_TOPLEVEL, NULL);
```

Comment:
note: the default constructor, [`gtk_window_new()`](https://developer.gnome.org/gtk2/stable/GtkWindow.html#gtk-window-new) has just those parameters, why not use it :)